components/shared/net/indexeddb_thread.rs

```diff
@@ -15,19 +15,55 @@ pub enum IndexedDBTxnMode {
 }
 
 // https://www.w3.org/TR/IndexedDB-2/#key-type
-#[derive(Clone, Debug, Deserialize, Serialize)]
+// FIXME:(arihant2math) Ordering needs to completely be reimplemented as per https://www.w3.org/TR/IndexedDB-2/#compare-two-keys
+#[derive(Clone, Debug, Deserialize, Serialize, PartialOrd, PartialEq)]
 pub enum IndexedDBKeyType {
-    Number(Vec<u8>),
-    String(Vec<u8>),
+    Number(f64),
+    String(String),
     Binary(Vec<u8>),
+    // FIXME:(arihant2math) Date should not be stored as a Vec<u8>
     Date(Vec<u8>),
-    // FIXME:(arihant2math) implment Array(),
+    Array(Vec<IndexedDBKeyType>),
+    // FIXME:(arihant2math) implment ArrayBuffer
 }
 
 // https://www.w3.org/TR/IndexedDB-2/#key-range
-#[derive(Clone, Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
 #[allow(unused)]
-pub enum IndexedDBKeyRange {}
+pub struct IndexedDBKeyRange {
+    pub lower: Option<IndexedDBKeyType>,
+    pub upper: Option<IndexedDBKeyType>,
+    pub lower_open: bool,
+    pub upper_open: bool,
+}
+
+impl From<IndexedDBKeyType> for IndexedDBKeyRange {
+    fn from(key: IndexedDBKeyType) -> Self {
+        IndexedDBKeyRange {
+            lower: Some(key.clone()),
+            upper: Some(key),
+            ..Default::default()
+        }
+    }
+}
+
+impl IndexedDBKeyRange {
+    // https://www.w3.org/TR/IndexedDB-2/#in
```

Comment:
make this a proper doccomment please

Reply:
The link seems to be correct to me, it links to "A key is in a key range if both of the following conditions are fulfilled:"

Reply:
Yes, I meant to make it a comment that rustdoc will recognize: `/// <https://www.w3.org/TR/IndexedDB-2/#in>`